Mathlib/Computability/RegularExpressions.lean

```diff
@@ -308,21 +319,20 @@ theorem star_rmatch_iff (P : RegularExpression α) :
         constructor
         · simp [hs, hsum]
         · intro t' ht'
-          cases' ht' with ht' ht'
-          · rw [ht']
-            exact ⟨by decide, ht⟩
-          · exact helem _ ht'
-    · rintro ⟨S, hsum, helem⟩
+          cases' ht' with ht' ht' -- porting note: last ht' doesn't take, need to `rename_i` below
+          · simp only [ne_eq, not_false_iff, true_and, rmatch]
+            exact ht
+          · rename_i ht'; exact helem t' ht'
```

Comment:
Can you make this work by using the structured cases syntax?

Reply:
works!